src/app/detail/[task_id]/[user_type]/page.tsx

```diff
@@ -144,6 +151,11 @@ export default async function TaskDetailPage({
                       const data = await getSignedUrlUpload(token, fileName)
                       return data
                     }}
+                    getSignedUrlFile={async (filePath: string) => {
+                      'use server'
+                      const data = await getSignedUrlFile(token, filePath)
+                      return data
+                    }}
```

Comment:
can we use this directly in the `TaskEditor` imported from `action.tsx` instead of passing it as a prop? For that to work make sure `use server` is placed on top of the action file.

Reply:
I think we need to pass token and every other requests are handled like this, the server component passes the request function to the client components. Using this directly in TaskEditor will need some extra token prop to be passed which I think is unnecessary because other service functions are not using this. I think for better consistency, we should not change this. what do you think?

Reply:
I think we need to work on making things better. I think there is no point of passing this as a prop when it can be used directly imported from the actions. There are a few concerns - 
1. `getSignedUrlFile` function is written in this file where as this function is more about action. Let's put this in the actions file. 
2. We are using the return `data` from this function in the `TaskEditor` which is a right approach. We can directly import `getSignedUrlFile` in the `TaskEditor` from the actions file and use the return. 
3. We have access to `token` from the redux store.

I think it is necessary to directly import all other actions as well. @arpandhakal